src/server/json_family.cc

```diff
@@ -588,8 +589,101 @@ OpResult<long> OpClear(const OpArgs& op_args, string_view key, string_view path)
   return clear_items;
 }
 
+// Returns string vector that represents the pop out values.
+OpResult<vector<OptString>> OpArrPop(const OpArgs& op_args, string_view key, string_view path,
+                                     int index) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptString> vec;
+  auto cb = [&](const string& path, json& val) {
+    if (!val.is_array() || val.empty()) {
+      vec.emplace_back(nullopt);
+      return;
+    }
+
+    size_t removal_index;
+    if (index < 0) {
+      int temp_index = index + val.size();
+      removal_index = abs(temp_index);
+    } else {
+      removal_index = index;
+    }
+
+    if (removal_index >= val.size()) {
+      removal_index %= val.size();  // rounded to the array boundaries.
+    }
+
+    size_t i = 0;
+    for (auto it = val.array_range().begin(); it != val.array_range().end(); it++) {
```

Comment:
there is no way to just access an element at index `i`? you must do the iteration yourself?

Reply:
This was the first thing that I've checked and unfortunately the library does not support access to individual members within an array.

Reply:
well, it that case, maybe `auto it = std::next(val.array_range().begin(), removal_index);'` seems like a better choice.

Reply:
Fixed